### PR TITLE
New Login and Hud Action bar, tweaks

### PR DESCRIPTION
**Pull request by tcape**
* New login scene
* Hud Action bar with abilities and potions

### CODE REVIEW

File: stc_game/Assets/Scripts/HUD/PotionButton.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    public PotionCooldown potionCooldown;
+
+    private void Start()
+    {
+        potionCooldown = parent.GetComponentInChildren<PotionCooldown>();
+    }
+
+    public void UsePotion()
+    {
+        var stats = FindObjectOfType<Hero>().characterStats.stats;
```
Comment (dixonyant) [inline stc_game/Assets/Scripts/HUD/PotionButton.cs:21]: We'll need to make this into a Hero Controller reference down the road

File: stc_game/Assets/Scripts/HUD/PotionCooldown.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class PotionCooldown : MonoBehaviour
+{
+    [SerializeField] public Transform parent;
```
Comment (dixonyant) [inline stc_game/Assets/Scripts/HUD/PotionCooldown.cs]: this serializeField should be above the variable

File: stc_game/Assets/Scripts/HUD/PotionCooldown.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+public class PotionCooldown : MonoBehaviour
+{
+    [SerializeField] public Transform parent;
+    private float cooldown;
+    public Image image;
+    public float startTime;
+
+    private void Awake()
+    {
+        cooldown = parent.GetComponent<PotionButton>().cooldown;
```
Comment (dixonyant) [inline stc_game/Assets/Scripts/HUD/PotionCooldown.cs:16]: Is there a way to get these by Id? down the road those won't be scalable to only reference by type

File: stc_game/Assets/Scripts/ReviveUI/ReviveController.cs [hunk 4]
```diff
@@ -9,20 +9,25 @@ public class ReviveController : MonoBehaviour
...
+        sceneController = SceneController.Instance;
     }
 
     private void Update()
     {
-        if (SceneController.Instance.currentSceneName.Equals(GameStrings.Scenes.TownScene))
-            reviveAtEntrance.SetActive(false);
-        else
-            reviveAtEntrance.SetActive(true);
+        if (sceneController)
```
Comment (dixonyant) [inline stc_game/Assets/Scripts/ReviveUI/ReviveController.cs]: why is this in the Update function?

File: stc_game/Assets/Scripts/SceneControl/PersistentScene/PersistentScene.cs [hunk 5]
```diff
@@ -37,54 +80,23 @@ private void Start()
         questWindowUI = GameObject.FindObjectOfType<QuestWindowUI>();
         QuestManager.instance.questWindowUI = PersistentScene.Instance.questWindowUI;
         reviveController = FindObjectOfType<ReviveController>();
+        
+        //actionBar = FindObjectOfType<ActionBarController>();  => This returns null every time, not sure why
```
Comment (dixonyant) [inline stc_game/Assets/Scripts/SceneControl/PersistentScene/PersistentScene.cs]: what is the action bar?

File: stc_game/Assets/Scripts/SceneControl/PersistentScene/PersistentScene.cs [hunk 6]
```diff
@@ -37,54 +80,23 @@ private void Start()
...
-                                               "Savagry",
-                                               "Cleave",
-                                               "Sprint"
-                                            }
-                                            );
+        if (UserService.Instance.user._id != "")
+        {
+            userService.GetUser(AuthService.Instance.authUser.sub);
+        }
+        else // for now so we can start from persistent scene without having to login
```
Comment (dixonyant) [inline stc_game/Assets/Scripts/SceneControl/PersistentScene/PersistentScene.cs:93]: yeah this is my bad, I need to get a local storage for the user or a mock user data that can be loaded here before the scene

File: stc_game/Assets/Scripts/Targets/DestinationController.cs [hunk 7]
```diff
@@ -1,37 +1,32 @@
...
     public Vector3 floorOffset;
-
-    
+    private StateController controller;
 
     // Start is called before the first frame update
     void Start()
     {
-        hero = GameObject.FindGameObjectWithTag("Player");
+        hero = GameObject.FindGameObjectWithTag("Player").GetComponent<Hero>();
```
Comment (dixonyant) [inline stc_game/Assets/Scripts/Targets/DestinationController.cs:16]: again, we'll get this into a service, just note for now all the files we'll have to touch to refactor for the hero data

File: stc_game/Assets/Scripts/HUD/PotionButton.cs (hunk 1 above)
Comment (tcape) [reply stc_game/Assets/Scripts/HUD/PotionButton.cs:21]: ok

File: stc_game/Assets/Scripts/HUD/PotionCooldown.cs (hunk 2 above)
Comment (tcape) [reply stc_game/Assets/Scripts/HUD/PotionCooldown.cs]: ok

File: stc_game/Assets/Scripts/ReviveUI/ReviveController.cs (hunk 4 above)
Comment (tcape) [reply stc_game/Assets/Scripts/ReviveUI/ReviveController.cs]: oops, changed it to subscribe to the AfterSceneLoad event instead

File: stc_game/Assets/Scripts/SceneControl/PersistentScene/PersistentScene.cs (hunk 5 above)
Comment (tcape) [reply stc_game/Assets/Scripts/SceneControl/PersistentScene/PersistentScene.cs]: Action bar is the bar on the hud that holds all the ability buttons

File: stc_game/Assets/Scripts/SceneControl/PersistentScene/PersistentScene.cs (hunk 6 above)
Comment (tcape) [reply stc_game/Assets/Scripts/SceneControl/PersistentScene/PersistentScene.cs:93]: np, this was just a quick temporary fix

File: stc_game/Assets/Scripts/Targets/DestinationController.cs (hunk 7 above)
Comment (tcape) [reply stc_game/Assets/Scripts/Targets/DestinationController.cs:16]: ok

File: stc_game/Assets/Scripts/HUD/PotionCooldown.cs (hunk 3 above)
Comment (tcape) [reply stc_game/Assets/Scripts/HUD/PotionCooldown.cs:16]: I'm not sure. This just grabs the value from the parent gameobject in the hierarchy. And the only way to set the value is in the editor for now. I just wanted to get a prototype of these potions to work.

File: stc_game/Assets/Scripts/SceneControl/PersistentScene/PersistentScene.cs (hunk 5 above)
Comment (dixonyant) [reply stc_game/Assets/Scripts/SceneControl/PersistentScene/PersistentScene.cs]: can this line be removed altogether then?

Comment (tcape) [reply stc_game/Assets/Scripts/SceneControl/PersistentScene/PersistentScene.cs]: yeah. I wanted to set that reference here, and was going to investigate why it was always null, but haven't done that yet. It's set from the ActionBarController now.

Comment (dixonyant) [reply stc_game/Assets/Scripts/SceneControl/PersistentScene/PersistentScene.cs]: oh ok

**Review by dixonyant — APPROVED** [review]

**Pull request merged**